dotCMS/src/test/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptorTest.java

```diff
@@ -0,0 +1,136 @@
+     * Given Scenario: the feature flag is off, so the matcher test should be not called
+     * ExpectedResult: The test matcher should be not called
+     */
+    @Test
+    public void test_intercept_feature_flag_turn_off() throws IOException {
+
+        final HostWebAPI hostWebAPI = Mockito.mock(HostWebAPI.class);
+        final AppsAPI appsAPI  = Mockito.mock(AppsAPI.class);
+        final WhiteBlackList whiteBlackList  = Mockito.mock(WhiteBlackList.class);
+        final AtomicBoolean isTurnedOn = new AtomicBoolean(false); // turn off the feature flag
+        final TestMatcher testMatcher = new TestMatcher();
+        final AnalyticsTrackWebInterceptor interceptor = new AnalyticsTrackWebInterceptor(
+                hostWebAPI, appsAPI, whiteBlackList, isTurnedOn, testMatcher);
+        final HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+        final HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        interceptor.intercept(request, response);
+
+        Assert.assertFalse("The test matcher should be not called, ff is off", testMatcher.wasCalled());
+    }
+
+    /**
+     * Method to test: AnalyticsTrackWebInterceptor#intercept
+     * Given Scenario: the feature flag is on but not config, so the matcher test should be not called
+     * ExpectedResult: The test matcher should be not called
+     */
+    @Test
+    public void test_intercept_feature_flag_turn_on_and_no_analytics_app() throws IOException, DotDataException, DotSecurityException {
+
+        final HostWebAPI hostWebAPI = Mockito.mock(HostWebAPI.class);
+        final AppsAPI appsAPI  = Mockito.mock(AppsAPI.class);
+        final WhiteBlackList whiteBlackList  = Mockito.mock(WhiteBlackList.class);
+        final AtomicBoolean isTurnedOn = new AtomicBoolean(true); // turn on the feature flag
+        final HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+        final HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+        final TestMatcher testMatcher = new TestMatcher();
+        final Host currentHost = new Host();
+
+        Mockito.when(hostWebAPI.getCurrentHostNoThrow(request)).thenReturn(currentHost);
+        Mockito.when(appsAPI.getSecrets(AnalyticsApp.ANALYTICS_APP_KEY,
+                true, currentHost, APILocator.systemUser())).thenReturn(Optional.empty()); // no config
+
+        final AnalyticsTrackWebInterceptor interceptor = new AnalyticsTrackWebInterceptor(
+                hostWebAPI, appsAPI, whiteBlackList, isTurnedOn, testMatcher);
+
+        interceptor.intercept(request, response);
+
+        Assert.assertFalse("The test matcher should be not called, no config set", testMatcher.wasCalled());
+    }
+
+    /**
+     * Method to test: AnalyticsTrackWebInterceptor#intercept
+     * Given Scenario: the feature flag is on and there is config, so the matcher test should be  called
+     * ExpectedResult: The test matcher should be  called
```

Comment:
linting to remove a few extra spaces
```suggestion
     * Given Scenario: the feature flag is on and there is config, so the matcher test should be called
     * ExpectedResult: The test matcher should be called
```